### ISSUE TITLE
dAppeteer with Jest not working in Headless mode

### ISSUE THREAD

**Issue by asimpk** [ChainSafe/dappeteer#270]
**Describe the bug**
I am facing an issue while running dAppeteer with jest in headless mode, in setup.js launch runs successfully but setupMetaMask doesn't work properly, similar issue is dappeteer.bootstrap() runs perfectly without headless mode

**To Reproduce**
1. Follow the doc to run DAppteer with Jest  https://github.com/ChainSafe/dappeteer/blob/master/docs/JEST.md
2. const browser = await launch(puppeteer, PUPPETEER_CONFIG), add the Headless: true option, BTW doc needs to be updated. 
3. run the tests

**Logs**

```
Determining test suites to run...
        Running tests on MetaMask version v10.23.0
        Flask version: false,
        Headless: true
```

**Expected behavior**
Tests should run, and logs should be seen to get show progress or give error

**System:**
- OS: windows
- OS version windows 10
- NodeJs version v16.16.0
- dAppeteer version ^4.2.0
- testing framework jest
- testing framework version 27.5.1

**Additional context**
A similar bug is seen when we run script without jest from as in README.md;
https://github.com/ChainSafe/dappeteer
const { metaMask, browser } = await dappeteer.bootstrap({
      browser: "chrome",
      headless: true
  });

**Comment by BeroBurny** [comment]
Thanks for pointing issue.
Can you tell me which automation framework you use and what version?

**Comment by asimpk** [comment]
Sorry, I'm not using any framework right now, I first tried to use synpress `https://github.com/Synthetixio/synpress`  but not using it and any other

**Comment by asimpk** [comment]
> Thanks for pointing issue. Can you tell me which automation framework you use and what version?

**Comment by BeroBurny** [comment]
try to install [playwright](https://playwright.dev/) or [puppeteer](https://pptr.dev/) and then run tests

**Comment by asimpk** [comment]
> try to install [playwright](https://playwright.dev/) or [puppeteer](https://pptr.dev/) and then run tests

Yeah I'm using "puppeteer": "^19.6.2", as that, was the requirement for setup

**Comment by BeroBurny** [comment]
it is confirmed that there is a bug, for now, the workaround is to use the playwright

<hr>

_observed notes_  

https://github.com/ChainSafe/dappeteer/blob/ee4a5136250a4b64d8045233227b9926cb203e75/src/setup/setupMetaMask.ts#L96-L113
for some reason, `targetcreated` is never triggered in puppeteer

cc @irubido @Lykhoyda

**Comment by BeroBurny** [comment]
simple repo for reproducing issue https://github.com/BeroBurny/dappeteer-jest

**Comment by asimpk** [comment]
> it is confirmed that there is a bug, for now, the workaround is to use the playwright
> 
> _observed notes_
> 
> https://github.com/ChainSafe/dappeteer/blob/ee4a5136250a4b64d8045233227b9926cb203e75/src/setup/setupMetaMask.ts#L96-L113
> 
> 
> for some reason, `targetcreated` is never triggered in puppeteer
> cc @irubido @Lykhoyda

Thanks @BeroBurny , by using playwright, it throws an exception from environment.js 
`if (!wsEndpoint) throw new Error('wsEndpoint not found');`
but in the case of puppeteer we get browser.wsEndpoint(),
 `ws://127.0.0.1:50927/devtools/browser/{id}`
